File: songs/php/05-extract-c.php
<?php
require_once(__DIR__.'/MySongs.php');

$songs = array_filter($songs, function ($song) {
  return isset($song['tags']);
});

$tagss = array_map(function ($song) {
  return $song['tags'];
}, $songs);

printf("%s\n", json_encode($tagss));

File: songs/perl/05-extract-c.pl
#!/usr/bin/perl
use 5.010;
use strict;
use warnings;

use File::Basename;
use lib dirname(__FILE__);

use MySongs;

my @tagss = ();

for (@MySongs::songs) { 
  if (grep /tags/, keys %$_) {
    push @tagss, $_->{'tags'};
  }
}

for (@tagss) {
  say join(":", @$_);
}
